webhook/bootstrapprovider_webhook.go

```diff
@@ -42,11 +43,33 @@ var _ webhook.CustomValidator = &BootstrapProviderWebhook{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *BootstrapProviderWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	provider, ok := obj.(*operatorv1.BootstrapProvider)
+	if !ok {
+		return fmt.Errorf("expected a BootstrapProvider object, got %T", obj)
+	}
+
+	if errList := validateProviderSpec(provider.Spec.ProviderSpec); errList != nil {
```

Comment:
Is it better to check the length of the error list rather than if it's non nil? Typically that would be the pattern for a list type